config/clusters/projectpythia/pythia-binder.values.yaml

```diff
@@ -23,7 +23,33 @@ jupyterhub:
     # The profile list in the common values files needs to be emptied
     # otherwise the launching will fail
     profileList: []
-    cmd: jupyter-lab
+    cmd:
+      - python3
+      - "-c"
+      - |
+        import os
+        import sys
+
+        try:
+            import jupyterlab
+            import jupyterlab.labapp
+            major = int(jupyterlab.__version__.split(".", 1)[0])
+        except Exception as e:
+            print("Failed to import jupyterlab: {e}", file=sys.stderr)
+            have_lab = False
+        else:
+            have_lab = major >= 3
+
+        if have_lab:
+            # technically, we could accept another jupyter-server-based frontend
+            print("Launching jupyter-lab", file=sys.stderr)
+            exe = "jupyter-lab"
+        else:
+            print("jupyter-lab not found, launching jupyter-notebook", file=sys.stderr)
+            exe = "jupyter-notebook"
+
```

Comment:
This is what binderhub does too. Otherwise, in images that don't have jupyter-lab installed, spawning a server will fail